lib/iris/fileformats/grib/_save_rules.py

```diff
@@ -200,13 +199,23 @@ def shape_of_the_earth(cube, grib):
 
     # Spherical earth.
     if ellipsoid.inverse_flattening == 0.0:
-        gribapi.grib_set_long(grib, "shapeOfTheEarth", 1)
-        gribapi.grib_set_long(grib, "scaleFactorOfRadiusOfSphericalEarth", 0)
+        # Set shapeOfTheEarth key based on value of spherical earth radius.
+        if int(ellipsoid.semi_major_axis) == 6367470:
+            gribapi.grib_set_long(grib, "shapeOfTheEarth", 0)
+        elif int(ellipsoid.semi_major_axis) == 6371229:
+            gribapi.grib_set_long(grib, "shapeOfTheEarth", 6)
+        else:
+            gribapi.grib_set_long(grib, "shapeOfTheEarth", 1)
+        gribapi.grib_set_long(grib,
+                              "scaleFactorOfRadiusOfSphericalEarth", 0)
         gribapi.grib_set_long(grib, "scaledValueOfRadiusOfSphericalEarth",
                               ellipsoid.semi_major_axis)
```

Comment:
It seems odd to be setting the radius even for shape 0 and 6 which have an implicit shape...?